File: go-ds-and-algorithms/sort/quick_sort_test.go
package sort

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestQuickSort(t *testing.T) {
	nums := []int{4, 2, 7, 1, 3}
	quickSort(nums, 0, len(nums)-1)
	assert.Equal(t, []int{1, 2, 3, 4, 7}, nums)
}

File: java-ds-and-algorithms/src/test/java/sort/QuickSortTest.java
package sort;

import org.junit.Assert;
import org.junit.Test;


/**
 * Usage: {@link QuickSort}测试 <br/>
 * Date: 2023/5/17 10:11 <br/>
 *
 * @author <a href="mailto:hanzhang2566@foxmail.com">hanzhang</a>
 */
public class QuickSortTest {

    @Test
    public void sortTest01() {
        int[] nums = new int[]{0, 5, 2, 1, 6, 3};
        new QuickSort(nums).sort(0, nums.length - 1);
        Assert.assertArrayEquals(new int[]{0, 1, 2, 3, 5, 6}, nums);
    }
}
